terra_notebook_utils/drs.py

```diff
@@ -46,7 +47,7 @@ def enable_requester_pays(workspace_name: Optional[str]=WORKSPACE_NAME,
     rawls_url = (f"https://rawls.dsde-{TERRA_DEPLOYMENT_ENV}.broadinstitute.org/api/workspaces/"
                  f"{workspace_namespace}/{encoded_workspace}/enableRequesterPaysForLinkedServiceAccounts")
     logger.info("Enabling requester pays for your workspace. This will only take a few seconds...")
-    access_token = gs.get_access_token()
+    access_token = get_terra_access_token()
```

Comment:
Thinking -- should we just avoid the `enable_requester_pays` entirely when the execution context is Azure? This is really a GCP-specific function..

Reply:
Yes, I think that makes sense. Even for Terra GCP, the only remaining case where enabling requester pays is necessary is AnVIL data hosted on Gen3, and even that is going away later this year, as described here #400

Reply:
@rtitle, this is now addressed/resolved in PR #402.

For example, in Terra Azure, when attempting to access a DRS URI entails requester pays:
```
jupyter@vm07858848f:~$ tnu drs copy drs://dg.ANV0/1b1ee6fc-6560-4b08-9c44-36d46bf4daf1 .
RequesterPaysNotSupported: Requester pays data access is not supported on the Azure platform. Cannot access: drs://dg.ANV0/1b1ee6fc-6560-4b08-9c44-36d46bf4daf1
```
The new code checks if requester pays is required for any of the given DRS URIs before proceeding to enable requester pays. If requester pays is required, and the platform is Azure, the error above is generated.

As a backup, the function to enable requester pays now includes an an assertion that the platform is not Azure:
```
def enable_requester_pays(workspace_name: Optional[str]=WORKSPACE_NAME,
                          workspace_namespace: Optional[str]=WORKSPACE_NAMESPACE):

    assert get_execution_context().execution_platform != ExecutionPlatform.AZURE, \
        "Requester pays data access is not supported on the Terra Azure platform."
```